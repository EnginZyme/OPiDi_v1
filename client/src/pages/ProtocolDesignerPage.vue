<template>
  <div class="main_container">
    <MainSidePanel />
    <router-view id="left_side_panel" name="LeftSidePanel" />
    <div class="secondary_container">
      <router-view id="top_nav_bar" name="TopNavBar" />
      <router-view id="display_view" name="DisplayView" />
    </div>
  </div>
</template>

<script>
import MainSidePanel from "../components/layout/MainSidePanel.vue";

/**
 * A component that renders the protocol designer page.
 */
export default {
  components: { MainSidePanel },
  name: "ProtocolDesignerPage",
  beforeCreate: function() {
    const userIsLoggedIn = this.$store.getters.userIsLoggedIn;

    if (!userIsLoggedIn)
      this.$router.push({
        name: "Authentication",
      });
  },
};
</script>

<style scoped>
.main_container {
  display: flex;
  flex-direction: row;
  align-items: stretch;
}
.secondary_container {
  display: flex;
  flex-direction: column;
  justify-items: stretch;
  width: 100%;
}
#display_view {
  display: grid;
  width: 100%;
  height: 100%;
  justify-content: stretch;
  align-content: stretch;
}
</style>
