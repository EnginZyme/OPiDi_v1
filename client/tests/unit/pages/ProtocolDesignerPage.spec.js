import { shallowMount, createLocalVue } from "@vue/test-utils"
import Vuex from "vuex";
import ProtocolDesignerPage from "@/pages/ProtocolDesignerPage"

const localVue = createLocalVue();

localVue.use(Vuex);

describe("shallowMounted ProtocolDesignerPage", () => {
    let getters;
    let store;
    let wrapper;
  
    beforeEach(() => {
      getters = {
        userIsLoggedIn: () => true,
      };
  
      store = new Vuex.Store({
        getters,
      });
  
      wrapper = shallowMount(ProtocolDesignerPage, {
        store,
        localVue,
      });
    });

    it("renders correctly", () => {
        expect(wrapper.element).toMatchSnapshot();
    })
})
