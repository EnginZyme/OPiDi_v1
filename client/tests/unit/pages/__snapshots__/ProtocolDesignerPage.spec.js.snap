// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallowMounted ProtocolDesignerPage renders correctly 1`] = `
<div
  class="main_container"
>
  <mainsidepanel-stub />
   
  <router-view
    id="left_side_panel"
    name="LeftSidePanel"
  />
   
  <div
    class="secondary_container"
  >
    <router-view
      id="top_nav_bar"
      name="TopNavBar"
    />
     
    <router-view
      id="display_view"
      name="DisplayView"
    />
  </div>
</div>
`;
